pkg/reconciler/subscription/subscription.go

```diff
@@ -201,6 +211,20 @@ func (r *Reconciler) resolveSubscriber(ctx context.Context, subscription *v1.Sub
 		if subscriber.Ref != nil {
 			subscriber.Ref.Namespace = subscription.Namespace
 		}
+
+		// Resolve the group
+		if subscriber.Ref != nil && feature.FromContext(ctx).IsEnabled(feature.KReferenceGroup) {
+			err := subscriber.Ref.ResolveGroup(r.crdLister)
```

Comment:
Perhaps we should do this logic inside `URIFromDestinationV1` ?

Reply:
Uhm maybe? Isn't it going to add too much concerns to `URIResolver`? I have no strong opinion about it...